packages/jellyfish-api-core/__tests__/category/governance/createCfp.test.ts

```diff
@@ -0,0 +1,138 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import BigNumber from 'bignumber.js'
+import { UTXO, ProposalStatus, ProposalType } from '../../../src/category/governance'
+import { RpcApiError } from '@defichain/jellyfish-api-core'
+
+describe('Governance without WalletCoinbaseMaturity', () => {
+  const container = new MasterNodeRegTestContainer(undefined, 'defi/defichain:HEAD-a61f615')
+  const client = new ContainerAdapterClient(container)
+  let payoutAddress: string
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+
+    payoutAddress = await container.call('getnewaddress')
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should not createCfp while still in Initial Block Download', async () => {
+    const data = {
+      title: 'Testing new community fund proposal',
+      amount: new BigNumber(100),
+      cycles: 2,
+      payoutAddress
+    }
+    const promise = client.governance.createCfp(data)
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('RpcApiError: \'Cannot create a cfp while still in Initial Block Download\', code: -10, method: createcfp')
+  })
+
+  it('should not createCfp with insufficient fund', async () => {
+    const data = {
+      title: 'Testing new community fund proposal',
+      amount: new BigNumber(100),
+      cycles: 2,
+      payoutAddress
+    }
+    await container.generate(1)
+    const promise = client.governance.createCfp(data)
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('RpcApiError: \'Insufficient funds\', code: -4, method: createcfp')
+  })
+})
+
+describe('Governance', () => {
+  const container = new MasterNodeRegTestContainer(undefined, 'defi/defichain:HEAD-a61f615')
+  const client = new ContainerAdapterClient(container)
+  let payoutAddress: string
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+
+    payoutAddress = await container.call('getnewaddress')
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should createCfp', async () => {
+    const data = {
+      title: 'Testing new community fund proposal',
+      amount: new BigNumber(100),
+      cycles: 2,
+      payoutAddress
+    }
+    const proposalTx = await client.governance.createCfp(data)
+    await container.generate(1)
```

Comment:
This should be correct. It confirms the execution of `createCfp` rpc. I think you should add `await container.generate(1)` after` await client.governance.createCfp(data)` for the rest of your code

Reply:
Ok. What about the throwing promises ?

Reply:
I think you should do it too.

Reply:
In that case, they don't throw anymore.
From what I can see in the other tests, they don't `await container.generate(1)` to check throwing promises as can be seen in :
`accountToAccount.test.ts` or 
`removePoolLiquidity.test.ts`.
Is there another way to do it ?

Reply:
Hmm, you are right. should not apply to promise